ui/src/lib/yjs/useWorkflowTabs.ts

```diff
@@ -27,7 +32,7 @@ export default ({
         id: w2.id as string,
         name: w2.name as string,
       })),
-    [rawWorkflows.length], // eslint-disable-line react-hooks/exhaustive-deps
+    [rawWorkflows.length, workflowName], // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Due to workflowName having rawworkflows as a dependency, this means when rawWorkflows changes (with each change), workflowName will update, which means workflows will update. So same behaviour we are tring to avoid with the rawWorkflows.length.